libs/clients/directorate-of-immigration/src/lib/directorateOfImmigrationClient.service.ts

```diff
@@ -0,0 +1,608 @@
+    private criminalRecordApi: CriminalRecordApi,
+    private optionSetApi: OptionSetApi,
+    private residenceAbroadApi: ResidenceAbroadApi,
+    private staticDataApi: StaticDataApi,
+    private studyApi: StudyApi,
+    private travelDocumentApi: TravelDocumentApi,
+
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) {}
+
+  private applicantApiWithAuth(auth: Auth) {
+    return this.applicantApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private applicantResidenceConditionApiWithAuth(auth: Auth) {
+    return this.applicantResidenceConditionApi.withMiddleware(
+      new AuthMiddleware(auth),
+    )
+  }
+
+  private applicationApiWithAuth(auth: Auth) {
+    return this.applicationApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private applicationAttachmentApiWithAuth(auth: Auth) {
+    return this.applicationAttachmentApi.withMiddleware(
+      new AuthMiddleware(auth),
+    )
+  }
+
+  private countryOfResidenceApiWithAuth(auth: Auth) {
+    return this.countryOfResidenceApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private criminalRecordApiWithAuth(auth: Auth) {
+    return this.criminalRecordApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private residenceAbroadApiWithAuth(auth: Auth) {
+    return this.residenceAbroadApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private staticDataApiWithAuth(auth: Auth) {
+    return this.staticDataApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private studyApiWithAuth(auth: Auth) {
+    return this.studyApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private travelDocumentApiWithAuth(auth: Auth) {
+    return this.travelDocumentApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  // Common:
+
+  async getCountries(): Promise<Country[]> {
+    try {
+      const res = await this.optionSetApi.apiOptionSetLookupTypeGet({
+        lookupType: LookupType.Countries,
+      })
+
+      return res.map((item) => ({
+        id: item.id?.toString() || '',
+        name: item.name || '',
+      }))
+    } catch (e) {
+      const errorMsg = `Error when trying to get countries from UTL`
+      this.logger.error(errorMsg, e)
+      throw new Error(errorMsg)
+    }
+  }
+
+  async getTravelDocumentTypes(): Promise<TravelDocumentType[]> {
+    try {
+      const res = await this.optionSetApi.apiOptionSetLookupTypeGet({
+        lookupType: LookupType.TravelDocumentTypes,
+      })
+
+      return res.map((item) => ({
+        id: item.id || 0,
+        name: item.name || '',
+      }))
+    } catch (e) {
+      const errorMsg = `Error when trying to get travel document types from UTL`
+      this.logger.error(errorMsg, e)
+      throw new Error(errorMsg)
+    }
+  }
+
+  // Citizenship:
+
+  async getOldCountryOfResidenceList(
+    auth: Auth,
+  ): Promise<CountryOfResidence[]> {
+    try {
+      const res = await this.countryOfResidenceApiWithAuth(
+        auth,
+      ).apiCountryOfResidenceGetAllGet()
+
+      return res.map((item) => ({
```

Comment:
Why the mapping of all response data from the UTL service instead of using the generated types from the openapi?

Reply:
I´m not familiar with a specific type strategy for the application system, I usually try to utilise the generated types if they are clean (and in english). You could check in with team Norda if they have more strict preference on this.